src/serializer.rs

```diff
@@ -143,6 +132,32 @@ impl<'a> ToCss for Token<'a> {
     }
 }
 
+fn to_hex_byte(value: u8) -> u8 {
+    match value {
+        0...9 => value + b'0',
+        _ => value - 10 + b'a',
+    }
+}
+
+fn hex_escape<W>(ascii_byte: u8, dest: &mut W) -> fmt::Result where W:fmt::Write {
+    let high = ascii_byte >> 4;
+    let b3;
+    let b4;
+    let bytes = if high > 0 {
+        let low = ascii_byte & 0x0F;
+        b4 = [b'\\', to_hex_byte(high), to_hex_byte(low), b' '];
+        &b4[..]
+    } else {
+        b3 = [b'\\', to_hex_byte(ascii_byte), b' '];
+        &b3[..]
+    };
+    dest.write_str(unsafe { str::from_utf8_unchecked(&bytes) })
```

Comment:
This could use `write_char`, I think, but not sure if worth it, this would be faster and it's trivially ascii so...

Reply:
When writing more than one byte/char it might be better to minimize the number of calls, but I haven’t measured.